indico/modules/events/editing/controllers/base.py

```diff
@@ -11,28 +11,30 @@
 from werkzeug.exceptions import Forbidden
 
 from indico.modules.events.contributions.controllers.display import RHContributionDisplayBase
-from indico.modules.events.controllers.base import RHEventBase
+from indico.modules.events.controllers.base import RHDisplayEventBase
 from indico.modules.events.editing.models.editable import Editable, EditableType
 from indico.modules.events.management.controllers.base import RHManageEventBase
+from indico.web.rh import RHProtected
 
 
-class RHEditingBase(RHEventBase):
+class RHEditingBase(RHDisplayEventBase):
     """Base class for editing RHs that don't reference an editable."""
 
     EVENT_FEATURE = 'editing'
 
     def _check_access(self):
-        RHEventBase._check_access(self)
-        # TODO: check for proper event/editing access (RHEventBase does NOT check event access)
+        RHDisplayEventBase._check_access(self)
         if not session.user:
             raise Forbidden
 
 
 class RHEditingManagementBase(RHManageEventBase):
+    """Base class for editing RHs that don't reference an editable."""
+
     EVENT_FEATURE = 'editing'
 
 
-class RHContributionEditableBase(RHContributionDisplayBase):
+class RHContributionEditableBase(RHContributionDisplayBase, RHProtected):
```

Comment:
What about adding a `RequireUserMixin` and using that here and in `RHEditingBase`? That way we don't need to inherit from two RH base classes, and still avoid having the `session.user` check in many different places..